nvc/extract.go

```diff
@@ -0,0 +1,125 @@
+package nvc
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+func ReadMagic(r io.Reader) error {
+	header := [8]byte{}
+	if _, err := io.ReadFull(r, header[:]); err != nil {
+		return err
+	}
+
+	if string(header[:]) != MAGIC {
+		return errors.New(".nvc signature not found")
+	}
+
+	return nil
+}
+
+func extractNVC(arcPath string, pathlist []string, outputDirectory string, extractUnknown bool) error {
```

Comment:
I think we're approaching the point where this function should be moved out of the `nvc` package (and into a cobra subcommand). This function is really application-level code; I don't think it belongs in this package.

That should be handled in a separate PR though.

Reply:
:boom: